src/java.base/share/classes/java/lang/constant/ConstantUtils.java

```diff
@@ -75,6 +75,60 @@ static String validateInternalClassName(String name) {
          return name;
      }
 
+    /**
+     * Validates the correctness of a binary package name. In particular checks for the presence of
+     * invalid characters in the name.
```

Comment:
It may be useful to note that the empty string are considered valid. 
Also add \@throws NPE; there's an implicit null check in checking the length.
Also in validateBinaryPackageName and validateModuleName.

Reply:
I'll add that notes, thanks.